python/pyarrow/_flight.pyx

```diff
@@ -474,6 +485,148 @@ cdef class GeneratorStream(FlightDataStream):
         return new CPyGeneratorFlightDataStream(self, self.schema, callback)
 
 
+cdef class ServerCallContext:
+    """Per-call state/context."""
+    cdef:
+        const CServerCallContext* context
+
+    def peer_identity(self):
+        """Get the identity of the authenticated peer.
+
+        May be the empty string.
+        """
+        return tobytes(self.context.peer_identity())
+
+    @staticmethod
+    cdef ServerCallContext wrap(const CServerCallContext& context):
+        cdef ServerCallContext result = \
+            ServerCallContext.__new__(ServerCallContext)
+        result.context = &context
+        return result
+
+
+cdef class ServerAuthReader:
+    """A reader for messages from the client during an auth handshake."""
+    cdef:
+        CServerAuthReader* reader
+
+    cdef read(self):
```

Comment:
Is this working? If it's a `cdef` method, it's not callable from Python, right?

Reply:
Good point, I'll cover that with a test case as well.